binding.c

```diff
@@ -514,6 +517,44 @@ static const char* execute_hmac(
   return NULL;
 }
 
+static const char* execute_sign(
+  const unsigned char* key,
+  const int key_size,
+  const unsigned char* source,
+  const int source_size,
+  unsigned char* target,
+  int* target_size
+) {
+  RSA *rsa = NULL;
+  BIO *keybio = BIO_new_mem_buf((void*)key, key_size);
+  if (keybio == NULL) {
+    return "key buffer allocation failed";
+  }
+  rsa = PEM_read_bio_RSAPrivateKey(keybio, &rsa, NULL, NULL);
```

Comment:
Creating the RSA private key struct seems like it should be performed a single. If you need to sign content repeatedly with the same key, this is extra work.

Reply:
Yes, it might be better to have a synchronous `key(buffer, offset, size)` method which accepts a private or public key , parses it as an OpenSSL read-only key, and returns it as a napi external, which can be passed back in later multiple times as an argument to `sign()` or `verify()` respectively.

By the way, what is the ms overhead of parsing a thousand keys?

Reply:
The other advantage of doing this would be removing all the `RSA_free()` calls in `execute_sign()`.

Reply:
I haven't actually benchmarked it. I just saw it in the code. My use case is to repeatedly sign JWTs with the same key thousands of times. So I'm looking for any performance wins I can.